online-shop/src/app/components/containers/products-form/products-form.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { take } from 'rxjs';
+import { getProduct, updateProduct } from 'src/app/modules/shared/state/actions/product.actions';
+import { selectAdminRole } from 'src/app/modules/shared/state/selectors/auth.selectors';
+import { selectCurrentProduct } from 'src/app/modules/shared/state/selectors/product.selectors';
+import { AppState } from 'src/app/modules/shared/state/state/app.state';
+import { Product } from 'src/app/modules/shared/types/products.types';
+
+@Component({
+  selector: 'app-products-form',
+  template: `<ng-container *ngIf="selectedProduct$">
+  <app-products-form-view [product]="selectedProduct$ | async" (updateProduct)="updateProduct($event)"></app-products-form-view>
+</ng-container>`,
+})
+export class ProductsFormComponent implements OnInit {
+  selectedProduct$ = this.store.select(selectCurrentProduct);
+  admin: Boolean = false;
+
+  constructor(
+    private route: ActivatedRoute,
+    private store: Store<AppState>
+  ) { }
+
+  ngOnInit() {
+    this.store.select(selectAdminRole).pipe(take(1)).subscribe(data => this.admin = data)
```

Comment:
done . same reason (Here i wanted to take the value of adminRole only one time because it cant be changed unless the logout button is pressed and another user signs in (in this case the add component will be loaded again)